pilot/pkg/config/kube/gateway/gateway_collection.go

```diff
@@ -212,18 +389,57 @@ func (p RouteParents) fetch(ctx krt.HandlerContext, pk parentKey) []*parentInfo
 		}
 	}
 	return slices.Map(krt.Fetch(ctx, p.gateways, krt.FilterIndex(p.gatewayIndex, pk)), func(gw Gateway) *parentInfo {
-		return &gw.parentInfo
+		return &gw.ParentInfo
 	})
 }
 
 func BuildRouteParents(
 	gateways krt.Collection[Gateway],
 ) RouteParents {
 	idx := krt.NewIndex(gateways, "parent", func(o Gateway) []parentKey {
-		return []parentKey{o.parent}
+		return []parentKey{o.Parent}
 	})
 	return RouteParents{
 		gateways:     gateways,
 		gatewayIndex: idx,
 	}
 }
+
+func detectListenerPort(l gatewayx.ListenerEntry) (gatewayx.PortNumber, error) {
```

Comment:
nit for clarity since we use port to refer to so many different data structures
```suggestion
func detectListenerPortNumber(l gatewayx.ListenerEntry) (gatewayx.PortNumber, error) {
```